### ISSUE TITLE
Liquorix Freezes on boot in VM

### ISSUE THREAD

**Issue by Batcastle** [zen-kernel/zen-kernel#148]
Running the latest version of Liquorix kernel, getting freezing like in Issue #141 

The `nothreadirqs` fix from that issue does fix it but it's not a permanent solution. Is there some way to fix this more permanently?

**Comment by damentz** [comment]
This may be a matter of which software and extensions you use to run your VM.  I'm already personally using Liquorix in a VM and on the server hosting https://liquorix.net.

Can you provide the output of `inxi -b` on the host OS?  Also, are you using VirtualBox, VMware, libvirtd, etc?  And which versions?

Also, after #141 was closed, the kernel produced afterwards disabled forced threaded IRQs.  This doesn't mean that _no_ IRQs use threads, just only enables threaded IRQs if supported.  By keeping `nothreadirqs` in your boot parameters, you might still be causing problems with your VM.

**Comment by Batcastle** [comment]
I'm using VirtualBox. None of the VB drivers are loaded into the VM (intentionally) and it's running version 6.0.4

Output of `inxi -b`:
```
System:    Host: batcastle-Nitro-AN515-53 Kernel: 5.0.0-17.1-liquorix-amd64 x86_64
           bits: 64
           Desktop: Xfce 4.12.3 Distro: Ubuntu 18.04.2 LTS
Machine:   Device: laptop System: Acer product: Nitro AN515-53 v: V1.22 serial: N/A
           Mobo: CFL model: Freed_CFS v: V1.22 serial: N/A
           UEFI: Insyde v: V1.22 date: 10/31/2018
Battery    BAT1: charge: 44.7 Wh 100.0% condition: 44.7/48.9 Wh (91%)
CPU:       Quad core Intel Core i5-8300H (-MT-MCP-) speed/max: 800/4000 MHz
Graphics:  Card-1: Intel Device 3e9b
           Card-2: NVIDIA GP107M [GeForce GTX 1050 Ti Mobile]
           Display Server: x11 (X.Org 1.19.6 )
           drivers: modesetting,nvidia (unloaded: fbdev,vesa,nouveau)
           Resolution: 1920x1080@60.02hz
           OpenGL: renderer: GeForce GTX 1050 Ti/PCIe/SSE2
           version: 4.6.0 NVIDIA 430.14
Network:   Card-1: Intel Wireless-AC 9560 [Jefferson Peak] driver: iwlwifi
           Card-2: Realtek RTL8111/8168/8411 PCIE Gigabit Ethernet Controller
           driver: r8169
Drives:    HDD Total Size: 256.1GB (83.4% used)
Info:      Processes: 437 Uptime: 4 min Memory: 2578.0/7815.9MB
           Client: Shell (bash) inxi: 2.3.56 
```

After Issue #141 was closed, I was having the same issue as before. Even after I applied the update you pushed out (Yeah I know I should have made another issue. Sorry about that. XD) So I have have had to keep adding it to boot. Also the last time I started up the VM I had to had `nomodeset` as well but for now I am assuming it was just some random occurrence. Just documenting it here in case it does turn out to be something bigger.

**Comment by damentz** [comment]
Which virtualization extensions do you have enabled and are actively using (VT-x / VT-d)?  I see that you're running an Acer, I'm not sure how much control you have over those settings in your BIOS.  Maybe adjusting those two features might change your stability, although turning on VT-x will impact throughput / performance.

**Comment by Batcastle** [comment]
I upgraded Virtualbox, and while that upgrade was overdue, It didn't seem to affect the issue. 

I started out with both VT-x and VT-d enabled, cause that's what I normally have enabled in BIOS. I tried disabling one, then the other. It looks like I have to keep VT-x enabled at the very least, but it does get slightly better performance with VT-d enabled as well. But no visible stability improvements.

Admittedly, this is not my area of expertise (although I do want to learn more and get involved more, but I digress). But I honestly feel like the issue lies with the Guest Kernel. Ubuntu Flavours on version 19.04 are running the 5.0 kernel and most of them are running just fine in a VM. I understand there are vast differences between the Ubuntu generic kernel and the Liquorix kernel. But, since they are very close in version I imagine the differences are fewer than normal right now. Again, not my area of expertise so I could be very wrong. But it's the only thing I can think of.

**Comment by damentz** [comment]
Have you tried changing the chipset and graphics controller in VirtualBox?  I think by default, VirtualBox selects PIIX3, but on modern distributions, ICH9 works pretty well.  And for the graphics controller, there's now 3 options, VBoxVGA (the original I believe), VMSVGA (VMware compatible), and VBoxSVGA (VirtualBox's new graphics controller).

If trying these doesn't help, can you come up with a test scenario so I can try replicating it on my side?  And I want to stress again, I use virtual machines heavily for my own personal use and work, and I haven't had any boot issues (CentOS, Ubuntu, Debian, MX Linux, etc).  This is across two different technologies as well, VMware workstation (mostly for Windows), and VirtualBox (for linuxes).

**Comment by damentz** [comment]
@Batcastle I think I found what happened.  Some the commit I referenced in the last issue never actually made it to the `master` or `5.0` branches, thus never physically changing the file contents of the config files.

This time I double checked and it shows up properly in the master branch from the github UI, and `git log` as well run locally.  How this happened, I have no clue.  My suspicion is I hit a strange edge case and the commit is corrupted.  Maybe if you know more about git than me, you might know what's going on behind the scenes.

I'm working on pushing out a new release with the forced IRQ threading properly reverted.  Should be out in the next few hours.  I'll wait for you to confirm this helps before closing out the issue.

**Comment by damentz** [comment]
@Batcastle I believe I might have reproduced the problem you were having on an Ubuntu 19.04 installation using Liquorix in VirtualBox.  If I install the kernel after a fresh install with updated packages, the next boot will hang what seems like indefinitely.  If I update grub so that no options are passed (remove `quiet` and `splash`), after what seems like 1-2 minutes, the display manager comes up and I'm able to use the desktop.

There's a lot of things that could be related in the systemd log.  Running `journalctl -x` points to many things, from bluetooth timing out during initialization to gnome shell failing all types of assertions before running.  This points to the distribution being the problem because I already tried this configuration on MX Linux (basically Debian Stable + lots of backports).  Liquorix works great in a VM of this type with the same VirtualBox configuration, booting instantly and being usable within seconds.

**Comment by Batcastle** [comment]
I am getting ready to try the new version in the VM. The OS is Drauger OS (based on Xubuntu 18.04 with mostly graphical changes and using Liquorix by default). So I know Gnome wouldn't be causing any issues since it isn't installed. CUPS wouldn't be either since it has been disabled. Bluetooth may have a hand to play in it though. 

I am currently trying the "no options passed" method you mentioned. If neither this nor the new kernel version helps, I will try playing around with the chipset and the graphics controller in different combos. So far it is still frozen up with the message shown in the screenshot below displayed:

![screenshot_05-26-2019_14:31:07](https://user-images.githubusercontent.com/31315409/58385775-9106c100-7fc3-11e9-9808-6fb0ed26e242.png)

**Comment by Batcastle** [comment]
I just updated the kernel to the latest version and it does indeed work. I still had to add `nomodeset`, as I mentioned in an earlier comment. But that is a completely different issue for another day. Lol

Thank you so much for your help, and all the hard work you do! I greatly appreciate it!

**Comment by damentz** [comment]
That output you got from disabling all the kernel options is interesting.  This got me looking at the RCU configuration in Liquorix, and I've been running with RCU expert options for quite a while instead of letting the kernel determine the most optimal configuration.  Just in-case this is related, I removed it and pushed out a new release (should be in the PPA in a few hours): https://github.com/damentz/liquorix-package/commit/f8c4b07f9a3d89624eb4998676ddb18e075dbbed

Can you give the 5.0.0-19.2 release a shot to see if it fixes `rcu_preempt detected expedited stalls` message?

**Comment by Batcastle** [comment]
I tried the new kernel release and when I booted it with the kernel options disabled it didn't show the `rcu_preempt detected expedited stalls` message, so I take that as a success. Is there somewhere/something I should check, just in case?

**Comment by damentz** [comment]
You can try `journalctl -k | grep -i rcu` to see if anything missed the TTY. Now, I don't know if the presence of this message indicates anything bad or if it's just a benign informational message.  The previous configuration had RCU priority boosting, so maybe in the event that an RCU thread needs to have its priority boosted, a message is logged to the TTY or kernel log, whichever is available.

**Comment by Batcastle** [comment]
Here is the output in that the command you provided gives:

![RCU-output-log](https://user-images.githubusercontent.com/31315409/58513161-dc100800-816c-11e9-80e5-96e2610f8bad.png)

This is immediately after boot up. I barely understand what any of this means since I have no idea what an RCU and/or RCU thread is (And Google was not very helpful) but maybe it's more useful to you.

**Comment by damentz** [comment]
Huh, yup, looks like those messages are gone.  Hopefully that's a good thing.

I'm going to close out this issue.  The original fix was to disable forced IRQ threading.  Some how my commit didn't change any branch (maybe some fat fingering on command line?), and correctly reverting the forced IRQ threading configuration fixed the problem, despite needing `nomodeset` on an Ubuntu 18.04 base.

If you're still interested in solving the mystery requiring `nomodeset` on your distro, open up an issue on the https://github.com/damentz/liquorix-package repository and I'll look at it from there.  My thoughts are there's probably a unique interaction between MuQSS, high resolution timeout patches, Ubuntu's choice of userspace, and a specific missing patch / backport that Ubuntu enables in their distribution kernel.
